src/expr/src/vector_op/cast.rs

```diff
@@ -218,6 +219,145 @@ pub fn bool_out(input: bool) -> Result<String> {
     Ok(if input { "t".into() } else { "f".into() })
 }
 
+macro_rules! gen_cast_impl {
```

Comment:
Comment the difference between this `gen_cast_impl` and the one in `src/expr/src/expr/expr_unary.rs`: This one does the actual cast over an individual scalar, while the one in `expr_unary.rs` generates an `Expression`.